sweeper/sweep_fee_provider.go

```diff
@@ -0,0 +1,280 @@
+package sweeper
+
+import (
+	"context"
+	"errors"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+// Sweeper provides fee, fee rate and weight by confTarget.
+type Sweeper interface {
+	// GetSweepFeeDetails calculates the required tx fee to spend to
+	// destAddr. It takes a function that is expected to add the weight of
+	// the input to the weight estimator. It also takes a label used for
+	// logging. It returns also the fee rate and transaction weight.
+	GetSweepFeeDetails(ctx context.Context,
+		addInputEstimate func(*input.TxWeightEstimator) error,
+		destAddr btcutil.Address, sweepConfTarget int32, label string) (
+		btcutil.Amount, chainfee.SatPerKWeight, lntypes.WeightUnit,
+		error)
+}
+
+// SweepFeeProvider provides fee rate and fee for a loop sweep transaction.
+type SweepFeeProvider struct {
+	// Sweeper returns fee rate by confTarget.
+	Sweeper Sweeper
+
+	// DefaultSweepConfTargetDelta defines a safe distance in blocks until
+	// the sweep's expiration height. If the distance is less than or equal
+	// to this value, then more aggressive fee is used to make sure that
+	// HTLC is swept before the sweep expires and the counterparty can
+	// collect refund.
+	DefaultSweepConfTargetDelta int32
+
+	// UrgentSweepConfTarget is the block confirmation target that is used
+	// for fee calculation of the on-chain htlc sweep if we are close to
+	// expiry.
+	UrgentSweepConfTarget int32
+
+	// DefaultSweepConfTargetFactor is the default factor by which the fee
+	// calculation increases the calculated fee.
+	DefaultSweepConfTargetFactor float32
+
+	// UrgentSweepConfTargetFactor is the factor by which the fee
+	// calculation increases the fee which resulted from the
+	// UrgentSweepConfTarget confirmation target. This aims to ensure that
+	// the sweep transaction will be confirmed asap in times when there is
+	// elevated block space demand in the top fee band.
+	UrgentSweepConfTargetFactor float32
+
+	// DisableHighPrioLogic disables special treatment of sweeps larger than
+	// or equal to HighPrioSweepAmount.
+	DisableHighPrioLogic bool
```

Comment:
I think he high prio part could be completely removed.

Reply:
Done. Now all the confTarget and feeRate logic is in method GetConfTargetAndFeeRate.